frontend/pages/post/[id].js

```diff
@@ -285,12 +376,46 @@ class LongPost extends React.Component {
       this.updatePost(name, value)
     }
   }
+  createComment = comment => {
+    this.context.actions
+      .createComment({
+        content: comment,
+        author: this.state.authenticatedUser._id,
+        post: this.state.data._id
+      })
+      .then(response => {
+        M.toast({
+          html: 'Successfully Created Comment',
+          classes: 'rounded green'
+        })
+        this.setState(prevState => ({
+          comments: [...prevState.comments, response.data]
+        }))
+      })
+      .catch(err => {
+        M.toast({ html: 'Oops, Something Went Wrong', classes: 'rounded red' })
+        console.log(err)
+        //(err)
+      })
+  }
 
   render() {
-    const { author } = this.state.data
+    let author
+    let userId
+    if (this.state.data.author) {
+      author = this.state.data.author._id
+    }
+    if (this.state.authenticatedUser) {
+      userId = this.state.authenticatedUser._id
+    }
+    this.setState(prevState => {
+      comments: this.state.data.comments
+    })
```

Comment:
can also be (if you're not going to use `prevState`):
```js
this.setState( { comments: this.state.data.comments } )

Reply:
I tried that at first but I get a Maximum update depth exceeded error. Using prevState for some reason fixes that.